chore(surveys): add branching logic docs (#8886)
PostHog/posthog.com: contents/docs/surveys/conditional-questions.mdx

---
title: Conditional questions
sidebar: Docs
showTitle: true
availability:
  free: none
  selfServe: full
  enterprise: full
---

> It is useful to be able to display survey questions conditionally. For example, you may want to ask certain questions based on the responses received. PostHog offers several ways to do this:

Reviewer: ```suggestion
It is useful to be able to display survey questions conditionally. For example, you may want to ask follow-up questions based on a user's responses. PostHog offers several ways to do this:
```

<ProductScreenshot
    imageLight="https://res.cloudinary.com/dmukukwp6/image/upload/Screenshot_2024_07_03_at_14_09_29_210ff7ced4.png" 
    imageDark="https://res.cloudinary.com/dmukukwp6/image/upload/Screenshot_2024_07_03_at_14_10_37_7804a733b4.png"
    alt="Survey rating branching" 
    classes="rounded"
/>

By default, a question proceeds to the next question in the survey unless specified otherwise. If you want a question to be the last one, but it is not the final question in the list, you need to explicitly set the value for **After this question go to ** to __Confirmation message__ (if present) or __End__..

<ProductScreenshot
    imageLight="https://res.cloudinary.com/dmukukwp6/image/upload/Screenshot_2024_07_03_at_14_15_15_8d0d9053d9.png" 
    imageDark="https://res.cloudinary.com/dmukukwp6/image/upload/Screenshot_2024_07_03_at_14_16_21_683fb41cda.png"
    alt="Survey end branching" 
    classes="rounded"
/>

> ## Testing
> We currently don't have the ability to test the survey flow interactively via preview, but we will add this feature soon. In the meantime, we recommend testing the branching flows by first releasing the survey to yourself.

Reviewer: ```suggestion
## Testing

We currently don't have the ability to test the survey flow interactively via preview, but we will add this feature soon. In the meantime, we recommend testing the branching flows by first releasing the survey to yourself.
```
